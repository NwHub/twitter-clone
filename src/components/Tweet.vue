<template>
  <div class="user-profile">
    <div class="user-profile__user-panel">
      <h1>@{{ user.userName }}</h1>
      <div>
        <strong>followers: {{ followers }} </strong>
        <button @click="followerCount">follow</button>
      </div>
      <!-- 入力 -->
      <form class="user-profile__create-tweet-panel">
        <label class="newTweet">
          <strong>New Tweet</strong>
        </label>
        <textarea rows="4"></textarea>
        <div>
          <button>Tweet</button>
        </div>
      </form>
    </div>
  </div>
</template>

<script>
export default {
  name: "Tweet",
  data: () => {
    return {
      followers: 0,
      user: {
        userName: "みわ",
        firstName: "泰平",
        lastName: "三輪"
      }
    };
  },
  computed: {
    hogeComputed() {
      return "";
    }
  },
  methods: {
    followerCount() {
      this.followers++;
    }
  }
};
</script>
<style>
.user-profile {
  display: grid;
  grid-template-columns: 1fr 3fr;
  grid-gap: 50px;
  padding: 50px 5%;
}
.user-profile__user-panel {
  display: flex;
  flex-direction: column;
  padding: 20px;
  background-color: white;
  border-radius: 5px;
  border: 1px solid #dfe3e8;
  margin-bottom: auto;
}
h1 {
  margin: 0;
}
button {
  padding: 5px 20px;
  margin: auto 0;
  border-radius: 5px;
  border: none;
  background-color: deeppink;
  color: white;
  font-weight: bold;
}
/* ツイート */
.user-profile__tweets-wrapper {
  display: grid;
  grid-gap: 10px;
  margin-bottom: auto;
}
.user-profile__tweets-wrapper__tweet-item {
  padding: 20px;
  background-color: white;
  border-radius: 5px;
  border: 1px solid #dfe3e8;
  box-sizing: border-box;
  cursor: pointer;
  transition: all 0.25s ease;
}
.user-profile__tweets-wrapper__tweet-item:hover {
  transform: scale(1.1, 1.1);
}
</style>
