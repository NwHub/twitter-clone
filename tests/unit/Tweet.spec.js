import { mount } from "@vue/test-utils";
import Tweet from "@/components/Tweet.vue";

describe("Tweet.vue", () => {
  it("renders props.msg when passed", () => {
    const wrapper = mount(Tweet);
    expect(wrapper).toMatchSnapshot();
  });
  it("button click", async () => {
    const wrapper = mount(Tweet);
    await wrapper.find("button").trigger("click");
    expect(wrapper).toMatchSnapshot();
  });
});
