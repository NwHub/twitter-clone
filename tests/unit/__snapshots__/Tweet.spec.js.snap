// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tweet.vue button click 1`] = `
<div class="user-profile">
  <div class="user-profile__user-panel">
    <h1>@みわ</h1>
    <div><strong>followers: 1 </strong> <button>follow</button></div>
    <form class="user-profile__create-tweet-panel"><label class="newTweet"><strong>New Tweet</strong></label> <textarea rows="4"></textarea>
      <div><button>Tweet</button></div>
    </form>
  </div>
  <div class="user-profile__tweets-wrapper">
    <div class="user-profile__tweets-wrapper__tweet-item">
      <p>Tweet1</p>
    </div>
    <div class="user-profile__tweets-wrapper__tweet-item">
      <p>Tweet2</p>
    </div>
  </div>
</div>
`;

exports[`Tweet.vue renders props.msg when passed 1`] = `
<div class="user-profile">
  <div class="user-profile__user-panel">
    <h1>@みわ</h1>
    <div><strong>followers: 0 </strong> <button>follow</button></div>
    <form class="user-profile__create-tweet-panel"><label class="newTweet"><strong>New Tweet</strong></label> <textarea rows="4"></textarea>
      <div><button>Tweet</button></div>
    </form>
  </div>
  <div class="user-profile__tweets-wrapper">
    <div class="user-profile__tweets-wrapper__tweet-item">
      <p>Tweet1</p>
    </div>
    <div class="user-profile__tweets-wrapper__tweet-item">
      <p>Tweet2</p>
    </div>
  </div>
</div>
`;
